### ISSUE TITLE
[json-schema-validator] update to 2.2.0

### ISSUE THREAD

**Issue by SamVanheer** [microsoft/vcpkg#28028]
Library name: json-schema-validator

New version number: 2.2.0

Other information that may be useful (release notes, etc...)
https://github.com/pboettch/json-schema-validator/releases/tag/2.2.0

**Comment by cthulhu-irl** [comment]
Hi. after port-update in MR #29253, it doesn't build on android debug mode anymore. there are build errors:
```
[1/8] C:\devtools\Android\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=aarch64-none-linux-android21 --sysroot=C:/devtools/Android/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot  -IC:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src -isystem C:/playpod/build-client-pc-next-Android_Qt_6_4_1_Clang_arm64_v8a-Debug/vcpkg_installed/arm64-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -Wall -Wextra -Wshadow -MD -MT CMakeFiles/nlohmann_json_schema_validator.dir/src/smtp-address-validator.cpp.o -MF CMakeFiles\nlohmann_json_schema_validator.dir\src\smtp-address-validator.cpp.o.d -o CMakeFiles/nlohmann_json_schema_validator.dir/src/smtp-address-validator.cpp.o -c C:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src/smtp-address-validator.cpp
FAILED: CMakeFiles/nlohmann_json_schema_validator.dir/src/smtp-address-validator.cpp.o 
C:\devtools\Android\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=aarch64-none-linux-android21 --sysroot=C:/devtools/Android/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot  -IC:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src -isystem C:/playpod/build-client-pc-next-Android_Qt_6_4_1_Clang_arm64_v8a-Debug/vcpkg_installed/arm64-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -Wall -Wextra -Wshadow -MD -MT CMakeFiles/nlohmann_json_schema_validator.dir/src/smtp-address-validator.cpp.o -MF CMakeFiles\nlohmann_json_schema_validator.dir\src\smtp-address-validator.cpp.o.d -o CMakeFiles/nlohmann_json_schema_validator.dir/src/smtp-address-validator.cpp.o -c C:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src/smtp-address-validator.cpp
smtp-address-validator.cpp:39:2: error: constant expression evaluates to -32 which cannot be narrowed to type 'char' [-Wc++11-narrowing]
        -32, -19, -16, -12, 34, 45, 61, 63,
        ^~~

...

smtp-address-validator.cpp:43:20: note: insert an explicit cast to silence this issue
        61, 64, -62, -33, -31, -17, -15, -13,
                          ^~~
                          static_cast<char>( )
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
[2/8] C:\devtools\Android\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=aarch64-none-linux-android21 --sysroot=C:/devtools/Android/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot  -IC:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src -isystem C:/playpod/build-client-pc-next-Android_Qt_6_4_1_Clang_arm64_v8a-Debug/vcpkg_installed/arm64-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -Wall -Wextra -Wshadow -MD -MT CMakeFiles/nlohmann_json_schema_validator.dir/src/json-uri.cpp.o -MF CMakeFiles\nlohmann_json_schema_validator.dir\src\json-uri.cpp.o.d -o CMakeFiles/nlohmann_json_schema_validator.dir/src/json-uri.cpp.o -c C:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src/json-uri.cpp
[3/8] C:\devtools\Android\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=aarch64-none-linux-android21 --sysroot=C:/devtools/Android/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot  -IC:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src -isystem C:/playpod/build-client-pc-next-Android_Qt_6_4_1_Clang_arm64_v8a-Debug/vcpkg_installed/arm64-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -Wall -Wextra -Wshadow -MD -MT CMakeFiles/nlohmann_json_schema_validator.dir/src/json-schema-draft7.json.cpp.o -MF CMakeFiles\nlohmann_json_schema_validator.dir\src\json-schema-draft7.json.cpp.o.d -o CMakeFiles/nlohmann_json_schema_validator.dir/src/json-schema-draft7.json.cpp.o -c C:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src/json-schema-draft7.json.cpp
[4/8] C:\devtools\Android\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=aarch64-none-linux-android21 --sysroot=C:/devtools/Android/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot  -IC:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src -isystem C:/playpod/build-client-pc-next-Android_Qt_6_4_1_Clang_arm64_v8a-Debug/vcpkg_installed/arm64-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -Wall -Wextra -Wshadow -MD -MT CMakeFiles/nlohmann_json_schema_validator.dir/src/json-patch.cpp.o -MF CMakeFiles\nlohmann_json_schema_validator.dir\src\json-patch.cpp.o.d -o CMakeFiles/nlohmann_json_schema_validator.dir/src/json-patch.cpp.o -c C:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src/json-patch.cpp
[5/8] C:\devtools\Android\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=aarch64-none-linux-android21 --sysroot=C:/devtools/Android/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot  -IC:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src -isystem C:/playpod/build-client-pc-next-Android_Qt_6_4_1_Clang_arm64_v8a-Debug/vcpkg_installed/arm64-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -Wall -Wextra -Wshadow -MD -MT CMakeFiles/nlohmann_json_schema_validator.dir/src/string-format-check.cpp.o -MF CMakeFiles\nlohmann_json_schema_validator.dir\src\string-format-check.cpp.o.d -o CMakeFiles/nlohmann_json_schema_validator.dir/src/string-format-check.cpp.o -c C:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src/string-format-check.cpp
[6/8] C:\devtools\Android\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=aarch64-none-linux-android21 --sysroot=C:/devtools/Android/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot  -IC:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src -isystem C:/playpod/build-client-pc-next-Android_Qt_6_4_1_Clang_arm64_v8a-Debug/vcpkg_installed/arm64-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -Wall -Wextra -Wshadow -MD -MT CMakeFiles/nlohmann_json_schema_validator.dir/src/json-validator.cpp.o -MF CMakeFiles\nlohmann_json_schema_validator.dir\src\json-validator.cpp.o.d -o CMakeFiles/nlohmann_json_schema_validator.dir/src/json-validator.cpp.o -c C:/vcpkg/buildtrees/json-schema-validator/src/2.2.0-e87f98a2b7.clean/src/json-validator.cpp
ninja: build stopped: subcommand failed.
```
is it related and can this be solved in this package's port?

**Comment by SamVanheer** [comment]
This was implemented in #29253.
